### DISCUSSION TITLE
Platform MPI

### DISCUSSION THREAD

**Discussion by DavidR-Vortex** [OpenRadioss/OpenRadioss#1194] [Q&A]
Hi,

Can OpenRadioss be compiled with Platform MPI on Linux? I can only see references to Open and Intel MPI in the documentation,

Many thanks,
David

**Comment by laurent-altr** [comment]
Hi David, 
As of now, we do not provide a way to build OpenRadioss with Platform MPI. 
Nevertheless, the code itself should be compatible with Platform MPI.

May I ask why you would like to use Platform MPI instead of OpenMPI or Intel MPI?

Thank you, 
Laurent

**Reply by DavidR-Vortex** [reply]
Hi Laurent,

The code is going onto an existing HPC system which has Platform MPI in use for other CAE codes. It would be less risky & tidier to use an existing MPI installation than add another into the ecosystem.

Kind regards,
David

**Reply by roguic** [reply]
Hi @DavidR-Vortex ,

Radioss supported in the past PL MPI. Some mpi routines were not supported by PL MPI, we added a variable (_PLMPI) in the engine to bypass some unsupported mpi routines : 
![image](https://user-images.githubusercontent.com/103993014/223742368-aa432363-6286-4648-a837-5b3119ecc3fe.png)


You can generate your own binaries with PL MPI. You need to create for the engine ( the starter does not use MPI) a new cmake_linux64_plmpi_gf.txt or modify the cmake_linux64_gf.txt to use PL MPI. The modifications are minor : you need to define mpi_inc to "I/path_to_your_PLMPI_folder/inlcude" and mpi_lib to "-L/path_to_your_PLMPI_folder/lib -lmpi -lpmpi" and add the variable _PLMPI to the compiler flags
![image](https://user-images.githubusercontent.com/103993014/223744114-f6c322c0-df00-4caf-9968-903a9e7ffbc4.png)

Best,
Romain

**Reply by DavidR-Vortex** [reply]
Thank you Romain - I will give this a go

Kind regards,
David
